app/src/main/java/ru/spbhse/pocketmagic/Network.java

```diff
@@ -0,0 +1,305 @@
+
+        @Override
+        public void onRoomCreated(int i, @Nullable Room room) {
+            Log.d("Pocket Magic", "Room has been created");
+            NetworkController.showMessage("Room has been created");
+        }
+
+        @Override
+        public void onJoinedRoom(int i, @Nullable Room room) {
+            Log.d("Pocket Magic", "Joined to room");
+            NetworkController.showMessage("Joined to room");
+        }
+
+        @Override
+        public void onLeftRoom(int i, @NonNull String s) {
+            Log.d("Pocket Magic", "Left from room");
+            NetworkController.showAlert("Disconnected");
+        }
+
+        @Override
+        public void onRoomConnected(int i, @Nullable Room room) {
+            if (room == null) {
+                Log.wtf("Pocket Magic", "Null pointer as Room!");
+                return;
+            }
+            Log.d("Pocket Magic", "Connected to room");
+
+            Network.this.room = room;
+            if (i == GamesCallbackStatusCodes.OK) {
+                Log.d("Pocket Magic", "Connected");
+            } else {
+                Log.e("Pocket Magic", "Error while connecting");
+                return;
+            }
+
+            Games.getPlayersClient(NetworkController.getContext(), account)
+                    .getCurrentPlayerId()
+                    .addOnSuccessListener(new OnSuccessListener<String>() {
+                        @Override
+                        public void onSuccess(String myPlayerId) {
+                            Network.this.playerID = Network.this.room.getParticipantId(myPlayerId);
+                            Log.d("Pocket Magic", "Received playerID");
+                            NetworkController.startGame();
+                        }
+                    });
+        }
+    }
+
+    private InvitationCallback invitationCallback = new InvitationCallback() {
+
+        @Override
+        public void onInvitationReceived(@NonNull Invitation invitation) {
+            Log.wtf("Pocket Magic", "Got invitation!");
+        }
+
+        @Override
+        public void onInvitationRemoved(@NonNull String invitationId) {
+            Log.wtf("Pocket Magic", "Invitation removed");
+        }
+    };
+
+    private RoomConfig roomConfig;
+    private Room room;
+    private GoogleSignInAccount account;
+    private RealTimeMultiplayerClient client;
+    private String playerID;
+
+    public Network(GoogleSignInAccount account) {
+        this.account = account;
+        client = Games.getRealTimeMultiplayerClient(NetworkController.getContext(), account);
+        GamesClient gamesClient = Games.getGamesClient(NetworkController.getContext(), account);
+        gamesClient
+                .getActivationHint()
+                .addOnSuccessListener(new OnSuccessListener<Bundle>() {
+                    @Override
+                    public void onSuccess(Bundle hint) {
+                        if (hint != null) {
```

Comment:
A possible code simplification would be to invert the conditions:

if (hint == null) {
  return;
}
Invitation invitation =  hint.getParcelable(Multiplayer.EXTRA_INVITATION);
if (invitation == null || invitation.getInvitationId() == null) {
  return;
}
// do stuff


Highly nested code is typically much harder to follow.

Reply:
Fixed.